securityscanutils/securityscan.go

```diff
@@ -266,10 +306,47 @@ func (r *SecurityScanRepo) UploadSecurityScanToGithub(fileName, versionTag strin
 	}
 	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/code-scanning/sarifs", r.Owner, r.Repo)
 	res, err := req.Post(url, req.BodyJSON(sarifMetadata), header)
-	fmt.Printf("Response from API, uploading sarif %s: \n %s\n", fileName, res.String())
 	if err != nil || res.Response().StatusCode != 200 {
 		return eris.Wrapf(err, "error uploading sarif file to github, response: \n%s", res)
 	}
+	fmt.Printf("Response from API, uploading sarif %s: \n %s\n", fileName, res.String())
+	return nil
+}
+
+// Creates/Updates a Github Issue per image
+// The github issue will have the markdown table report of the image's vulnerabilities
+// example: https://github.com/solo-io/solo-projects/issues/2458
+func (r *SecurityScanRepo) CreateUpdateVulnerabilityIssue(ctx context.Context, client *github.Client, image, markdownScanFilePath string) error {
+	issueTitle := fmt.Sprintf("Security Alert: %s", image)
+	trivyLabels := &[]string{"trivy", "vulnerability"}
+	markdownScan, err := ioutil.ReadFile(markdownScanFilePath)
+	if err != nil {
+		return eris.Wrapf(err, "error reading file %s", markdownScanFilePath)
+	}
+	issueRequest := &github.IssueRequest{
+		Title:  github.String(issueTitle),
+		Body:   github.String(string(markdownScan)),
+		Labels: trivyLabels,
+	}
+	createNewIssue := true
+
+	for _, issue := range r.allGithubIssues {
+		// If issue already exists, update existing issue with new security scan
+		if strings.Contains(issue.GetTitle(), issueTitle) {
+			// Only create new issue if issue does not already exist
```

Comment:
Only create new issue if (issue does not already exist **|| existing issue is closed)**

Not sure if it's possible we'd have duplicate titles, but if we somehow do, we don't want to update a closed issue and not catch wind of the vulnerability

Reply:
We only get open issues from github, so if the existing issues don't contain the title, it doesn't exist or is closed